File: frontend/src/app/models/gamedataDTOs/WisdomSkillTreeDefinitionDTO.ts
import { WisdomGroupType } from "../../enums/WisdomGroupType";
import { WisdomGroupDefinitionDTO } from "./WisdomGroupDefinitionDTO";

export interface WisdomSkillTreeDefinitionDTO {
    wisdomGroups: Record<WisdomGroupType, WisdomGroupDefinitionDTO>
}

File: backend/src/main/java/com/langleon/dsobuildsim/wisdomskilltree/dto/definition/WisdomSkillTreeDefinitionDTO.java
package com.langleon.dsobuildsim.wisdomskilltree.dto.definition;

import com.langleon.dsobuildsim.wisdomskilltree.wisdomgroup.WisdomGroupType;

import java.util.EnumMap;

public record WisdomSkillTreeDefinitionDTO(
        EnumMap<WisdomGroupType, WisdomGroupDefinitionDTO> wisdomGroups
) {
}
